### ISSUE TITLE
reproducing results on ImageNet-V dataset

### ISSUE THREAD

**Issue by yjh576** [jameelhassan/PromptAlign#10]
Despite successfully reproducing results on other datasets, I am unable to replicate the results of AlignPrompt on ImageNet-V using your code.  I observe that the accuracy on ImageNet-V is almost zero, and it seems that the model is unable to converge. I confirm that I have not altered the code and have implemented it correctly. Could you kindly provide suggestions or assistance? Your assistance would be greatly appreciated.

**Comment by jameelhassan** [comment]
Can you share the results that you get for Imagenet-V?

**Comment by yjh576** [comment]
Here are some of my reported results, and the model seems to neither converge nor perform effectively.

Re-updating prompt initializations to current prompts.
=> Using native Torch AMP. Training in mixed precision.
number of test samples: 10000
Running for 64 Augmented views
Running for 1 TTA steps
Test: [  199/10000]     Time  0.879 ( 1.067)    Acc@1   0.00 (  1.00)   Acc@5   0.00 (  2.50)
Test: [  399/10000]     Time  0.891 ( 0.977)    Acc@1   0.00 (  0.75)   Acc@5   0.00 (  2.00)

**Comment by Bostoncake** [comment]
I have the same issue reproducing results for v2. Did you manage to come up with some solutions?

**Comment by Bostoncake** [comment]
@jameelhassan I wonder if you would provide some insights on reproducing results for the v2 dataset? Thanks a lot!

**Comment by jameelhassan** [comment]
Hi @yjh576 and @Bostoncake 
The issue is in the dataset class. Unfortunately, I do not have access to the computing resources till mid-July. Will check this and update the code soon.

**Comment by dhruv-sarkar** [comment]
@jameelhassan can you kindly provide more information regarding what exactly the issue is? I have been trying to reproduce the results on ImageNet-v2 but have not been successful.

**Comment by Iris1946** [comment]
hi @jameelhassan
I was wondering if the issue with the ImageNet-v2 dataset not being reproducible has been resolved. Could you please let me know when you might be planning to update the code? Thank you very much for your help.

**Comment by Bostoncake** [comment]
Hi @jameelhassan @yjh576 @Iris1946,
I think I have solved the problem in reproducing ImageNet-V2.

The problem lies in the class dataset naming in im-v2. If you prepare the dataset as instructed by the README, you will get something like:
![image](https://github.com/user-attachments/assets/e0a89564-8211-4d6d-ae83-b5bb1850c88b)

PromptAlign uses tpt_loader in https://github.com/jameelhassan/PromptAlign/blob/master/trainers/prompt_align.py#L578 to conduct test-time adaptation. That dataset is constructed via the default DataLoader in PyTorch. And by default, the classes in the tpt_loader is arranged like this:
<img width="1102" alt="42888ed2e6d91bc5c6666169874d38c" src="https://github.com/user-attachments/assets/8ae3e22e-d8d5-4540-824d-35f2b2b1751f">

So the solution is simply renaming all class folders under imagenetv2, here is a sample script for doing this:
```
import os

# Path to the directory containing the class folders
path = 'data/imagenetv2/imagenetv2-matched-frequency-format-val'

# List all folder names in the specified directory
for folder_name in os.listdir(path):
    # Check if the current item is a directory
    if os.path.isdir(os.path.join(path, folder_name)):
        # Create the new folder name with zero-padding to ensure 3 digits
        new_name = folder_name.zfill(3)
        # Full paths for the old and new folder names
        old_path = os.path.join(path, folder_name)
        new_path = os.path.join(path, new_name)
        # Rename the folder
        os.rename(old_path, new_path)
        print(f"Renamed '{folder_name}' to '{new_name}'")
```

I hope this solves the issues when reproducing ImageNet-V2!

**Comment by Bostoncake** [comment]
BTW, it is also necessary to change https://github.com/jameelhassan/PromptAlign/blob/master/datasets/imagenetv2.py#L87 to:
```
class_dir = os.path.join(image_dir, str(label).zfill(3))
```

**Comment by Iris1946** [comment]
@Bostoncake Thank you so much 😊!!! I can successfully reproduce the ImageNetV2 results on my machine.

**Comment by jameelhassan** [comment]
@Bostoncake I did not have the time or access to the original codebase to check and fix this. 
Thank you very much for solving it.
